### ISSUE TITLE
Issues with JSON fields, `z.any()` and relations

### ISSUE THREAD

**Issue by franky47** [CarterGrimmeisen/zod-prisma#52]
When a model has a JSON field, its relationship declaration will throw a type error.

```prisma
// Example schema

model User {
  id    Int    @id @default(autoincrement())
  meta  Json
  posts Post[]
}

model Post {
  id       Int  @id @default(autoincrement())
  authorId Int
  author   User @relation(fields: [authorId], references: [id])
}
```

The issue comes from the Json type resolving to `z.any()`, and the extended schema not mapping correctly to the extended interface due to some part being optional and some other being required.

Rather than the builtin `any`, the package could include some runtime helpers to resolve JSON types. See [Zod's documentation](https://github.com/colinhacks/zod#json-type) for implementation suggestions, though this _may_ cause some issues when trying to resolve the schema types using `z.typeOf<typeof modelSchema>`.

**Comment by franky47** [comment]
Note: it seems that it's not related to JSON, I also have this issue with a model with an optional Date.

**Comment by franky47** [comment]
Hey @CarterGrimmeisen, the fix you added in b0b4152 (injecting a recursive JSON schema) seems to work, do you have an ETA on a release ?
